nipoppy/extractors/fmriprep/run_FC.py

```diff
@@ -0,0 +1,261 @@
+
+	return FC
+
+def run_FC(
+          participant_id: str,
+          session_id: str,
+          fmriprep_dir,
+		  DKT_dir,
+          FC_dir,
+          brain_atlas_lst,
+		  confound_strategy,
+		  metric_lst,
+		  task,
+		  run,
+		  space,
+		  logger: logging.Logger,
+):
+	""" Assess functional connectivity using Nilearn"""
+	
+	logger.info("Running FC assessment...")
+	logger.info("-"*50)
+	
+	try:
+		for brain_atlas in brain_atlas_lst:
+			logger.info('******** running ' + brain_atlas)
+			### extract time series
+			func_file = f"{fmriprep_dir}/{participant_id}/ses-{session_id}/func/{participant_id}_ses-{session_id}_{task}_{run}_{space}_desc-preproc_bold.nii.gz"
+			dkt_atlas = f"{DKT_dir}/{participant_id}/ses-{session_id}/anat/{participant_id}_ses-{session_id}_run-{run}_{space[:-6]}_atlas-DKTatlas+aseg_dseg.nii.gz" # space[:-6] removes the '_res2' suffix
+			if brain_atlas=='DKT':
+				time_series, labels = extract_timeseries(func_file, brain_atlas, confound_strategy, dkt_atlas)
+			else:
+				time_series, labels = extract_timeseries(func_file, brain_atlas, confound_strategy)
+
+			### assess FC
+			FC = assess_FC(time_series, labels, metric_lst=metric_lst)
+
+			## save output 
+			folder = f"{FC_dir}/output/{participant_id}/ses-{session_id}/"
+			if not os.path.exists(folder):
+				os.makedirs(folder)
+			np.save(f"{folder}/{participant_id}_ses-{session_id}_{task}_{space}_FC_{brain_atlas}.npy", FC)
+	except Exception as e:
+		logger.error(f"FC assessment failed with exceptions: {e}")
+
+	logger.info(f"Successfully completed FC assessment for participant: {participant_id}")
+	logger.info("-"*75)
+	logger.info("")
+
+
+def run(participant_id: str,
+        global_configs,
+		FC_configs,
+        session_id: str,
+        output_dir: str,
+        logger=None):
+	""" Runs fmriprep command
+	"""
+	DATASET_ROOT = global_configs["DATASET_ROOT"]
+	FMRIPREP_VERSION = global_configs["PROC_PIPELINES"]["fmriprep"]["VERSION"]
+
+	confound_strategy = FC_configs["confound_strategy"]
+	metric_lst = FC_configs["metric_lst"]
+	brain_atlas_lst = FC_configs["brain_atlas_lst"]
+	task = FC_configs["task"]
+	run = FC_configs["run"]
+	space = FC_configs["space"]
+
+	if metric_lst is None:
+		metric_lst = ['correlation']
+	if brain_atlas_lst is None:
+		brain_atlas_lst = [
+			'schaefer_100', 'schaefer_200',
+			'schaefer_300', 'schaefer_400',
+			'schaefer_500', 'schaefer_600',
+			'schaefer_800', 'schaefer_1000'
+		]
+	if confound_strategy is None:
+		confound_strategy = 'none'
+
+	log_dir = f"{DATASET_ROOT}/scratch/logs/"
+
+	if logger is None:
+		log_file = f"{log_dir}/FC.log"
+		logger = my_logger.get_logger(log_file)
+
+	logger.info("-"*75)
+	logger.info(f"Using DATASET_ROOT: {DATASET_ROOT}")
+	logger.info(f"Using participant_id: {participant_id}, session_id:{session_id}")
+
+	if output_dir is None:
+		output_dir = f"{DATASET_ROOT}/derivatives/"
+
+	fmriprep_dir = f"{DATASET_ROOT}/derivatives/fmriprep/v{FMRIPREP_VERSION}/output"
```

Comment:
Eventually we will need to make this more dynamic since different versions of fMRIPrep have slightly different output directory structures, but for now this should be fine

One thing though is that in the Boutiques PR I am changing the version directory to not have the `v` prefix anymore, so we probably want to remove it here too. @nikhil153 what do you think?

Reply:
I think we should remove all the `v` prefix in a separate PR as it requires changes in all the datasets on disk. Alternatively, we can add support for both.

Reply:
So I should nothing in this PR regarding this?